### ISSUE TITLE
go modules require a newer git than is available on centos6

### ISSUE THREAD

**Issue by jscook2345** [apptainer/singularity#3478]
## Version of Singularity:

master

On centos6, `mconfig` currently fails because `go` is not able to find the `go modules` correctly for modules using `git tag`.

The solution seems to be to update the `git` version for the build environment. I don't think `git` is a hard requirement for the package itself.

Since our centos6 testing is for building an RPM, I think that this should be OK.

**Comment by jscook2345** [comment]
@DrDaveD @mem @GodloveD @ikaneshiro @jmstover

**Comment by DrDaveD** [comment]
How is this problem reproduced?  I just tried building the latest master branch on my centos6 VM and it built fine.

**Comment by jscook2345** [comment]
@DrDaveD 

It happens in all of the travis builds, currently. All the PRs have the failing check I think.

What is your `git version` and `go version` output for centos6? Both myself and @GodloveD were able to reproduce this with `vagrant`, `virtualbox`, and a centos6.10 VM.

**Comment by DrDaveD** [comment]
```
$ git version
git version 1.7.1
$ go version
go version go1.11.5 linux/amd64
$ rpm -q git
git-1.7.1-9.el6_9.x86_64
$ rpm -q golang
golang-1.11.5-1.el6.x86_64
```

**Comment by jscook2345** [comment]
@mem made a good point that modules should not be downloaded at build-time and things should be using the vendor directory instead. His fix for this issues is here: #3481
